packages/client/src/app/components/form-controls/control-types/calculation-box/calculation-box.tsx

```diff
@@ -0,0 +1,103 @@
+import type Tagify from '@yaireo/tagify';
+import { MixedTags } from '@yaireo/tagify/dist/react.tagify';
+
+import { Control } from '../../control';
+
+import { CalculationBoxWrapper } from './calculation-box.styles';
+
+import '@yaireo/tagify/dist/tagify.css';
+
+const CalculationBox: React.FC<ControlType<CalculationProperty>> = ({
+  value,
+  property,
+  errors,
+  updateValue,
+}) => {
+  const [calculationBoxValue, setCalculationBoxValue] = useState('');
+  const allFields = useSelector(fieldSelectors.all);
+  const handles = allFields
+    .filter((item) =>
+      [
+        'Solspace\\Freeform\\Fields\\Implementations\\NumberField',
+        'Solspace\\Freeform\\Fields\\Implementations\\TextField',
+        'Solspace\\Freeform\\Fields\\Implementations\\TextareaField',
+        'Solspace\\Freeform\\Fields\\Implementations\\DropdownField',
+        'Solspace\\Freeform\\Fields\\Implementations\\RadiosField',
+        'Solspace\\Freeform\\Fields\\Implementations\\HiddenField',
+        'Solspace\\Freeform\\Fields\\Implementations\\Pro\\InvisibleField',
+        'Solspace\\Freeform\\Fields\\Implementations\\Pro\\OpinionScaleField',
+        'Solspace\\Freeform\\Fields\\Implementations\\Pro\\RatingField',
+        'Solspace\\Freeform\\Fields\\Implementations\\Pro\\RegexField',
+      ].includes(item.typeClass)
+    )
+    .map((item) => item.properties.handle);
+
+  const generateValue = (value: string): string => {
+    return value.replace(
+      /field:([a-zA-Z0-9_]+)/g,
+      (_, variable) => `[[${variable}]]`
+    );
+  };
+
+  const onChange = useCallback(
+    (e: CustomEvent<Tagify.ChangeEventData<Tagify.TagData>>) => {
+      const getValue = e.detail.tagify.DOM.input.textContent;
+
+      updateValue(getValue);
+    },
+    []
+  );
+
+  useEffect(() => {
+    setCalculationBoxValue(generateValue(value));
+  }, []);
+
+  return (
+    <Control property={property} errors={errors}>
+      <CalculationBoxWrapper>
+        <MixedTags
+          autoFocus={false}
+          settings={{
+            pattern: /@|{/,
+            mixTagsInterpolator: ['[', ']'],
+            enforceWhitelist: true,
+            editTags: false,
+            pasteAsTags: true,
+            dropdown: {
+              enabled: 0,
+              position: 'text',
+              includeSelectedTags: true,
+            },
+            templates: {
+              tag: function (tagData) {
+                return `<tag title='${
+                  tagData.value
+                }' contenteditable='false' spellcheck="false" class='tagify__tag' ${this.getAttributes(
+                  tagData
+                )}>
+                <x title='remove tag' class='tagify__tag__removeBtn'></x>
+                  <div>
+                    <span class='tagify__tag-text'><span class='sr-only-value'>field:</span>${
+                      tagData.value
+                    }${' '}</span>
+                  </div>
+                </tag>`;
+              },
```

Comment:
Did eslint format this block of code?

Watch out for mixed single quotes and double quotes.

This looks much more cleaner and readable (but not tested):

```
tag: function (tagData) {
    return `
        <tag
            title="${tagData.value}"
            contenteditable="false"
            spellcheck="false"
            class="tagify__tag"
            ${this.getAttributes(tagData)}
        >
            <x title="remove tag" class="tagify__tag__removeBtn"></x>
            <div class="tagify__tag-text">
                <span class="sr-only-value">field:</span>${tagData.value}${' '}
            </div>
        </tag>
    `;
},
```

Also, do we need the `sr-only-value` class ?
... and `${' '}` to create a space?

Reply:
It's easier to get the value from textContent and convert it to a tag. So, I add the 'sr-only-value' & for the space - This is because I retrieve the value from textContent. If anyone forgets to add space between the operator and the tag, it will be added.